### ISSUE TITLE
Why I get lower recall when I query in space which partition greater than 1?

### ISSUE THREAD

**Issue by ABNER-1** [vearch/vearch#348]
Why I get lower recall when I query in space which partition greater than 1?

I set partition as 1 and 2.
However, when I query with the same code, I got lower recall when 2 partitions than 1 partition.

**Comment by ABNER-1** [comment]
Hello, thanks for open sourcing this great project.
I have debuged that question yesterday.
I found a mistake in the design for `SortValueMap` in the situation that exists more than one partition.

The bug code link in [client code](https://github.com/vearch/vearch/blob/80bed7cd53da326da7476351787704cff46f4890/client/client.go#L1228).

The problem is that when nq > 1, the `firstSortValue` which updated in the method `SearchFieldSortExecute` as `sortValueMap` may be replaced by more than one same id.

For example: I ask nq 2 and topk as 50. When id 1 in both query result and their score is different, the bug happens.

Look forward for a better vearch.
